recon.py

```diff
@@ -0,0 +1,329 @@
+"""
+A Proposal for reconciliation graph and reconciliation interface.
+"""
+from abc import ABC  # Abstract Base Classes
+from typing import List, Dict
+from enum import Enum
+
+__all__ = ['MappingNode', 'EventType', 'Event', 'Cospeciation', 'Duplication', 'Transfer', 'Loss',
+           'TipTip', 'Reconciliation', 'ReconGraph']
+
+class MappingNode:
+    """
+    MappingNode is a node in the reconciliation graph that maps
+    a parasite to a host.
+    """
+    def __init__(self, parasite_vertex: str, host_vertex: str):
+        self._parasite = parasite_vertex
+        self._host = host_vertex
+
+    @property
+    def parasite(self) -> str:
+        return self._parasite
+
+    @property
+    def host(self) -> str:
+        return self._host
+    
+    def __eq__(self, other):
+        if type(self) == type(other):
+            return self._parasite == other.parasite and self._host == other.host
+        return False
+    
+    def __hash__(self):
+        return hash((self._parasite, self._host))
```

Comment:
Do you need a hash function?  If _parasite and _host are strings, can you just use the tuple directly when you need a hash?

Reply:
If we don't define a hash function for the class, it will not work with `dict`.